test/jsonapi-document-structure-resource-fields.test.js

```diff
@@ -0,0 +1,454 @@
+/* eslint-env mocha */
+import { JSONPath } from 'jsonpath-plus';
+import { expect } from 'chai';
+
+// Rules to test
+import ruleset from '../rules/jsonapi-document-structure-resource-fields.js';
+
+// Helper Functions
+import { resolveRef } from './utils/refResolver.js';
+import { handleSpectralResults } from './utils/handleSpectralResults.js';
+import { processErrors } from './utils/processErrors.js';
+import { debugLog, debugError, debugInfo, debugDebug } from './utils/debugUtils.js';
+import { setupSpectralBeforeEach } from './utils/setupSpectralBeforeEach.js';
+
+// OpenAPI Documents
+import invalidApiDocumentAttributesIdNotRequired from './docs/errors/resourceObjectFields/invalidApiDocumentAttributesIdNotRequired.js';
+import invalidApiDocumentAttributesTypeNotRequired from './docs/errors/resourceObjectFields/invalidApiDocumentAttributesTypeNotRequired.js';
+import invalidApiDocumentRelationshipsIdNotRequired from './docs/errors/resourceObjectFields/invalidApiDocumentRelationshipsIdNotRequired.js';
+import invalidApiDocumentRelationshipsTypeNotRequired from './docs/errors/resourceObjectFields/invalidApiDocumentRelationshipsTypeNotRequired.js';
+
+import validApiDocument from './docs/validApiDocument.js';
```

Comment:
Please remove line 21 as it is unnecessary and does not contribute to the functionality. 

You are currently using the global method for the `validApiDocument`